crmsh/utils.py

```diff
@@ -3214,4 +3094,35 @@ def parse_user_at_host(s: str):
     else:
         return s[:i], s[i+1:]
 
+
+def file_is_empty(file: str) -> bool:
+    return os.stat(file).st_size == 0
+
+
+def get_open_method(infile):
+    """
+    Get the appropriate file open method based on the file extension
+    """
+    file_type_open_dict = {
+        "gz": gzip.open,
+        "bz2": bz2.open,
+        "xz": lzma.open
+    }
+    file_ext = infile.split('.')[-1]
+    return file_type_open_dict.get(file_ext, open)
+
+
+def read_from_file(infile: str) -> str:
+    """
+    Read content from a file
+    """
+    _open = get_open_method(infile)
+    try:
+        with _open(infile, 'rt', encoding='utf-8', errors='replace') as f:
+            data = f.read()
+    except Exception as err:
+        logger.error("When reading file \"%s\": %s", infile, str(err))
+        return ""
+
+    return to_ascii(data)
```

Comment:
As the file is open in text mode, `f.read()` returns `str` and `to_ascii` is not needed.